File: modules/core/misc/java/test/PointTest.java
package org.opencv.test.core;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

import org.junit.Test;
import org.opencv.core.Point;
import org.opencv.core.Rect;
import org.opencv.test.OpenCVTestCase;

public class PointTest extends OpenCVTestCase {

    private Point p1;
    private Point p2;

    @Override
    public void setUp() throws Exception {
        super.setUp();

        p1 = new Point(2, 2);
        p2 = new Point(1, 1);
    }

    @Test
    public void testClone() {
        Point truth = new Point(1, 1);
        Point dstPoint = truth.clone();
        assertEquals(truth, dstPoint);
    }

    @Test
    public void testDot() {
        double result = p1.dot(p2);
        assertEquals(4.0, result, 0);
    }

    @Test
    public void testEqualsObject() {
        boolean flag = p1.equals(p1);
        assertTrue(flag);

        flag = p1.equals(p2);
        assertFalse(flag);
    }

    @Test
    public void testHashCode() {
        assertEquals(p1.hashCode(), p1.hashCode());
    }

    @Test
    public void testInside() {
        Rect rect = new Rect(0, 0, 5, 3);
        assertTrue(p1.inside(rect));

        Point p2 = new Point(3, 3);
        assertFalse(p2.inside(rect));
    }

    @Test
    public void testPoint() {
        Point p = new Point();

        assertNotNull(p);
        assertEquals(0.0, p.x, 0);
        assertEquals(0.0, p.y, 0);
    }

    @Test
    public void testPointDoubleArray() {
        double[] vals = { 2, 4 };
        Point p = new Point(vals);

        assertEquals(2.0, p.x, 0);
        assertEquals(4.0, p.y, 0);
    }

    @Test
    public void testPointDoubleDouble() {
        p1 = new Point(7, 5);

        assertNotNull(p1);
        assertEquals(7.0, p1.x, 0);
        assertEquals(5.0, p1.y, 0);
    }

    @Test
    public void testSet() {
        double[] vals1 = {};
        p1.set(vals1);
        assertEquals(0.0, p1.x, 0);
        assertEquals(0.0, p1.y, 0);

        double[] vals2 = { 6, 10 };
        p2.set(vals2);
        assertEquals(6.0, p2.x, 0);
        assertEquals(10.0, p2.y, 0);
    }

    @Test
    public void testToString() {
        String actual = p1.toString();
        String expected = "{2.0, 2.0}";
        assertEquals(expected, actual);
    }

}

File: modules/core/misc/objc/test/PointTest.swift
//
//  PointTest.swift
//
//  Created by Giles Payne on 2020/01/31.
//

import XCTest
import OpenCV

class PointTest: OpenCVTestCase {

    let p1 = Point2d(x: 2, y: 2)
    let p2 = Point2d(x: 1, y: 1)

    func testClone() {
        let truth = Point2d(x: 1, y: 1)
        let dstPoint = truth.clone()
        XCTAssertEqual(truth, dstPoint);
    }

    func testDot() {
        let result = p1.dot(p2);
        XCTAssertEqual(4.0, result)
    }

    func testEqualsObject() {
        var flag = p1 == p1
        XCTAssert(flag)

        flag = p1 == p2
        XCTAssertFalse(flag)
    }

    func testHashCode() {
        XCTAssertEqual(p1.hash(), p1.hash())
    }

    func testInside() {
        let rect =  Rect2d(x: 0, y: 0, width: 5, height: 3)
        XCTAssert(p1.inside(rect))

        let p2 = Point2d(x: 3, y: 3)
        XCTAssertFalse(p2.inside(rect))
    }

    func testPoint() {
        let p = Point2d()

        XCTAssertNotNil(p)
        XCTAssertEqual(0.0, p.x)
        XCTAssertEqual(0.0, p.y)
    }

    func testPointDoubleArray() {
        let vals:[Double] =  [2, 4]
        let p = Point2d(vals: vals as [NSNumber])

        XCTAssertEqual(2.0, p.x);
        XCTAssertEqual(4.0, p.y);
    }

    func testPointDoubleDouble() {
        let p1 = Point2d(x: 7, y: 5)

        XCTAssertNotNil(p1)
        XCTAssertEqual(7.0, p1.x);
        XCTAssertEqual(5.0, p1.y);
    }

    func testSet() {
        let vals1:[Double] = []
        p1.set(vals: vals1 as [NSNumber])
        XCTAssertEqual(0.0, p1.x)
        XCTAssertEqual(0.0, p1.y)

        let vals2 = [ 6, 10 ]
        p2.set(vals: vals2 as [NSNumber])
        XCTAssertEqual(6.0, p2.x)
        XCTAssertEqual(10.0, p2.y)
    }

    func testToString() {
        let actual = "\(p1)"
        let expected = "Point2d {2.000000,2.000000}"
        XCTAssertEqual(expected, actual)
    }

}
